client/platform/desktop/frontend/components/BulkImportDialog.vue

```diff
@@ -0,0 +1,195 @@
+      required: true,
+    },
+  },
+  setup(props, ctx) {
+    // Map imports to include generated "id" field, used in rendering.
+    const imports = ref(props.importData.map((im) => {
+      const cloned = cloneDeep(im);
+      cloned.jsonMeta.id = (Math.random() + 1).toString(36).substring(2);
+
+      return cloned;
+    }));
+
+    // The dataset import currently being configured
+    const currentImport = ref<DesktopMediaImportResponse>();
+
+    // Selected state and selected imports maintain the ordering of `imports`
+    const selectedState = ref(imports.value.map(() => true));
+    const selectedImports = computed(() => imports.value.filter((_, index) => selectedState.value[index]));
+    function updateSelected(val: DesktopMediaImportResponse[]) {
+      selectedState.value = imports.value.map((im) => val.includes(im));
+    }
+
+    function stripId(item: DesktopMediaImportResponse) {
+      const cloned = cloneDeep(item);
+      cloned.jsonMeta.id = '';
+      return item;
+    }
+
+    function finalizeImport() {
+      const finalImports = imports.value.filter((im) => selectedImports.value.includes(im)).map((im) => stripId(im));
+      ctx.emit('finalize-import', finalImports);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function updateImportConfig(oldItem: DesktopMediaImportResponse, newItem: DesktopMediaImportResponse) {
+      const itemIndex = imports.value.indexOf(oldItem);
+
+      // Need to modify the imports array ref to contain the new item
+      const newArray = clone(imports.value);
+      newArray.splice(itemIndex, 1, newItem);
+      imports.value = newArray;
+
+      currentImport.value = undefined;
+    }
+
+    function formatPath(item: DesktopMediaImportResponse) {
+      let path = item.jsonMeta.originalBasePath;
+
+      if (item.jsonMeta.originalVideoFile !== '') {
+        path = `${path}/${item.jsonMeta.originalVideoFile}`;
+      }
+
+      return path;
+    }
+
+    return {
+      updateSelected,
+      updateImportConfig,
+      formatPath,
+      imports,
+      headers,
+      selectedImports,
+      currentImport,
+      finalizeImport,
+    };
+  },
+});
+</script>
+
+<template>
+  <v-card outlined class="import-card" style="overflow-x: hidden;">
+    <v-card-title class="text-h5">
+      Bulk Import (Selecting {{ selectedImports.length }} of {{ imports.length }})
+    </v-card-title>
+
+    <v-dialog :value="currentImport !== undefined" width="800">
+      <ImportDialog
+        v-if="currentImport !== undefined"
+        :import-data="currentImport"
+        :embedded="true"
+        @abort="currentImport = undefined"
+        @finalize-import="updateImportConfig(currentImport, $event)"
+      />
+    </v-dialog>
+
+    <v-data-table
+      :value="selectedImports"
+      :items="imports"
+      item-key="jsonMeta.id"
+      :headers="headers"
+      show-select
+      disable-sort
+      @input="updateSelected"
```

Comment:
Believe you still need to set some sort of max width on the data table:

https://github.com/user-attachments/assets/16b1780e-4546-4214-91c2-e3bb8d16f4b7

Reply:
A specific `max-width` style wasn't necessary with the fix I implemented, but it still prevents overflow on all screen sizes (wraps text instead).